### ISSUE TITLE
Duplicate type registration errors when enabling field groups only on specific posts

### ISSUE THREAD

**Issue by benknight** [wp-graphql/wp-graphql-acf#198]
My queries are still executing but when I enable debugging I see a huge list of "duplicate type" errors, which seems like my custom post type "component" is having its GQL types registered more than once.

See errors in example server GQL response: https://gist.github.com/benknight/baab7b0842694bff54e096835d727803

Here's how the post type is being registered:

```php
function coracle__register_post_type() {
  register_post_type('component', [
    'exclude_from_search' => true,
    'graphql_single_name' => 'component',
    'graphql_plural_name' => 'components',
    'labels' => [
      'edit_item' => 'Edit Component',
      'name' => 'Components',
      'new_item' => 'New Component',
      'singular_name' => 'Component',
    ],
    'menu_icon' => 'dashicons-screenoptions',
    'menu_position' => 20,
    'public' => true,
    'rewrite' => [
      'slug' => 'browse',
    ],
    'show_ui' => true,
    'show_in_graphql' => true,
    'show_in_menu' => true,
    'show_in_nav_menus' => false,
    'supports' => ['custom-fields', 'title', 'revisions'],
  ]);
}

add_action('init', 'coracle__register_post_type');
```

Versions:

![image](https://user-images.githubusercontent.com/786691/103606280-93275480-4f48-11eb-8269-00002dfd44e4.png)

**Comment by benknight** [comment]
Wrong repo.

**Comment by benknight** [comment]
Re-opening, seems to be an issue with this ACF plugin (disabling plugin gets rid of the duplicate type registration errors).

**Comment by benknight** [comment]
This might be a duplicate of https://github.com/wp-graphql/wp-graphql-acf/issues/189

**Comment by benknight** [comment]
Temporary solution (as described here https://github.com/wp-graphql/wp-graphql-acf/issues/189#issuecomment-751748160) is to assign the field group based on post type as opposed to specific posts.

**Comment by jasonbahl** [comment]
I've reproduced this with the following location rules: 

![Screen Shot 2021-03-05 at 2 45 26 PM](https://user-images.githubusercontent.com/1260765/110176980-6e712100-7dc1-11eb-81e3-078df60fba2b.png)

The issue here is that the field group is created and registered to the `Post` type in the Schema from the first rule, then the 2nd rule tries to register the field group to a specific post, and the plugin attempts to register it to the `Post`. 

We've been discussing changing how location rules are handled, and instead of inferring location rules and mapping to the Schema the plan is to have a more explicit UI where users can define the specific Type(s) in the Schema where  an ACF Field group should be available.

**Comment by jasonbahl** [comment]
This is resolved by #250. 

I can now create a field group that contains a "Link" field type, and assign it to Posts and Pages:

![Screen Shot 2021-04-20 at 2 25 12 PM](https://user-images.githubusercontent.com/1260765/115459539-6855c900-a1e4-11eb-8669-b83068921d86.png)

Then query the field and get no debug message in the debug logs. No more `"You cannot register duplicate Types to the Schema. The Type 'ACF_Link' already exists in the Schema. Make sure to give new Types a unique name."` message. 

![Screen Shot 2021-04-20 at 2 25 02 PM](https://user-images.githubusercontent.com/1260765/115459537-67bd3280-a1e4-11eb-9eff-06517de302a9.png)
